packages/driver/src/multi-domain/domain_fn.ts

```diff
@@ -49,73 +30,82 @@ export const handleDomainFn = (cy: $Cy, specBridgeCommunicator: SpecBridgeDomain
 
     // Set the state ctx to the runnable ctx to ensure they remain in sync
     cy.state('ctx', cy.state('runnable').ctx)
+
+    // Stability is always false when we start as the page will always be
+    // loading at this point
+    cy.isStable(false, 'multi-domain-start')
   }
 
-  specBridgeCommunicator.on('run:domain:fn', async ({ data, fn, state, isDoneFnAvailable = false }: { data: any[], fn: string, isDoneFnAvailable: boolean, state: {}}) => {
-    reset(state)
+  const doneEarly = () => {
+    cy.queue.stop()
 
-    let fnWrapper = `(${fn})`
+    // we only need to worry about doneEarly when
+    // it comes from a manual event such as stopping
+    // Cypress or when we yield a (done) callback
+    // and could arbitrarily call it whenever we want
+    const p = cy.state('promise')
 
-    if (isDoneFnAvailable) {
-      // stub out the 'done' function if available in the primary domain
-      // to notify the primary domain if the done() callback is invoked
-      // within the spec bridge
-      const done = (err = undefined) => {
-        doneEarly()
+    // if our outer promise is pending
+    // then cancel outer and inner
+    // and set canceled to be true
+    if (p && p.isPending()) {
+      cy.state('canceled', true)
+      cy.state('cancel')()
+    }
 
-        // signal to the primary domain that done has been called and to signal that the command queue is finished in the secondary domain
-        specBridgeCommunicator.toPrimaryError('done:called', { err })
-        specBridgeCommunicator.toPrimary('queue:finished')
+    // if a command fails then after each commands
+    // could also fail unless we clear this out
+    cy.state('commandIntermediateValue', undefined)
 
-        return null
-      }
+    // reset the nestedIndex back to null
+    cy.state('nestedIndex', null)
```

Comment:
what is the nestedIndex?

Reply:
See https://github.com/cypress-io/cypress/blob/75c87502a2ca88fa76c8c5bb5f52626e0f70b08f/packages/driver/src/cypress/cy.ts#L1141